sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -359,12 +360,18 @@ static bool getUint32PropAsBool(const RTDeviceBinaryImage &Img,
   return Prop && (DeviceBinaryProperty(Prop).asUint32() != 0);
 }
 
-static int getUint32PropAsInt(const RTDeviceBinaryImage &Img,
-                              const char *PropName) {
+static const char *getUint32PropAsOptStr(const RTDeviceBinaryImage &Img,
+                                         const char *PropName) {
   pi_device_binary_property Prop = Img.getProperty(PropName);
+  std::stringstream ss;
   if (!Prop)
-    return -1;
-  return DeviceBinaryProperty(Prop).asUint32();
+    return (const char *)(ss.str().c_str());
```

Comment:
I think you're using the result of `c_str` past the lifetime of the object owning it. Why are you even using plain C interface here (I mean raw `const char *` return type)?

Reply:
Good point. Modified this to return std::string and it is a lot simpler design. Thanks